### PR TITLE
Promote operator version to v1beta1

### PR DESCRIPTION
**Pull request by bcrochet**
The API for the operator is stable. Stable enough to at least be beta.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
Make the operator's API beta.
```

### CODE REVIEW

File: deploy/crds/nmstate.io_nmstates_crd.yaml [hunk 1]
```diff
@@ -5,14 +5,14 @@ metadata:
 spec:
   group: nmstate.io
   names:
-    kind: NMstate
```
Comment (phoracek) [inline deploy/crds/nmstate.io_nmstates_crd.yaml:8]: You need to rebase the PR seems like.

File: deploy/crds/nmstate.io_v1beta1_nmstate_cr.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+apiVersion: nmstate.io/v1beta1
```
Comment (phoracek) [inline deploy/crds/nmstate.io_v1beta1_nmstate_cr.yaml:1]: Once you rebase, would you please fix it in https://github.com/nmstate/kubernetes-nmstate/blob/master/hack/prepare-release.sh#L54 as well?

File: deploy/crds/nmstate.io_nmstates_crd.yaml (hunk 1 above)
Comment (phoracek) [reply deploy/crds/nmstate.io_nmstates_crd.yaml:8]: Aha, https://github.com/nmstate/kubernetes-nmstate/pull/521 is not merged yet. Sorry

**Comment by qinqon** [comment]
@bcrochet can you rebase ?

File: pkg/apis/nmstate/v1alpha1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -64,65 +64,6 @@ func (in ConditionList) DeepCopy() ConditionList {
 	return *out
 }
 
-// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
```
Comment (qinqon) [inline pkg/apis/nmstate/v1alpha1/zz_generated.deepcopy.go:67]: Don't we have to maintain this, so we can still manage old versions ?

Comment (phoracek) [reply pkg/apis/nmstate/v1alpha1/zz_generated.deepcopy.go:67]: For the operator, I'd rather assume there is nobody upgrading from the previous. For the handler, we will need to kepe backwards compatibility. Here it is not that important (IMO).

Comment (qinqon) [reply pkg/apis/nmstate/v1alpha1/zz_generated.deepcopy.go:67]: fair enough

**Comment by phoracek** [comment]
@bcrochet seems like we have a rebase conflict:

```
./cluster/clean.sh: line 30: syntax error near unexpected token `<<<'
```

**Comment by qinqon** [comment]
@bcrochet let's rebase again

**Comment by phoracek** [comment]
/override pull-kubernetes-nmstate-e2e-handler-k8s

The fail is not realted.

**Pull request merged**